pkg/cmd/adm/register_member.go

```diff
@@ -120,59 +117,206 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+
+		// update the secret label
+		labels := kubeConfigSecret.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		labels[toolchainv1alpha1.ToolchainClusterLabel] = toolchainClusterName
+		kubeConfigSecret.Labels = labels
+
+		// update the kubeconfig data
+		kubeConfigSecret.StringData = map[string]string{
+			"kubeconfig": string(generatedKubeConfigFormatted),
+		}
+
+		return nil
+	})
+
 	if err != nil {
-		return nil, errs.Wrapf(err, "unable to get add-script.sh")
-	}
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return nil, fmt.Errorf("unable to get add-script.sh - response status %s", resp.Status)
+		return err
 	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil {
-			term.Printlnf(err.Error())
+	term.Println("Secret successfully reconciled")
+	term.Printlnf("operation", op)
+
+	// TODO -- temporary logic
+	// The creation of the toolchaincluster is just temporary until we implement https://issues.redhat.com/browse/KUBESAW-44,
+	// the creation logic will be moved to the toolchaincluster_resource controller in toolchain-common and will be based on the secret created above.
+	//
+	// create/update toolchaincluster on the clusterJoinTo
+	term.Printlnf("creating ToolchainCluster representation of %s in %s:", joiningClusterType, clusterJoinToName)
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: toolchainClusterName, Namespace: clusterJoinToOperatorNamespace}}
+	op, err = controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToClient, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
 		}
-	}()
-	// Create the file
-	file, err := os.CreateTemp("", "add-cluster-*.sh")
+		labels["namespace"] = joiningOperatorNamespace
+		if joiningClusterType == "member" {
+			labels["cluster-role.toolchain.dev.openshift.com/tenant"] = ""
+		}
+
+		toolchainClusterCR.Labels = labels
+		toolchainClusterCR.Spec.APIEndpoint = joiningClusterAPIEndpoint
+		toolchainClusterCR.Spec.SecretRef.Name = secretName
+		if insecureSkipTLSVerify {
+			toolchainClusterCR.Spec.DisabledTLSValidations = []toolchainv1alpha1.TLSValidation{toolchainv1alpha1.TLSAll}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+	term.Println("Toolchaincluster successfully reconciled")
+	term.Printlnf("operation", op)
+	// -- end temporary logic
+
+	return err
+}
+
+func secretName(SANamespacedName runtimeclient.ObjectKey, joiningOperatorNamespace string, joiningClusterName string) string {
+	secretName := SANamespacedName.Name + "-" + joiningOperatorNamespace + "-" + joiningClusterName
+	return secretName
+}
+
+func newRestClient(kubeConfigPath string) (*rest.RESTClient, error) {
+	restClientConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	restClientConfig.Timeout = 60 * time.Second
```

Comment:
Why are we overriding these? If we need to do it, it should be documented here why we're overriding the kubeconfig provided to us.

Reply:
I think this was a copy paste left-over. Removed in https://github.com/kubesaw/ksctl/pull/53/commits/b1b829351f35c9c7078ce31c4a1e7bb1d528e212 

The other part here https://github.com/kubesaw/ksctl/pull/53/files#diff-5c2ece44983cae536e1e2cd48a7ee24657235790b6a5b7c35c1adbe848d2330eR249-R252 is needed otherwise the rest client doesn't work and fails with an error.